docs/DeveloperGuide.md

```diff
@@ -1,34 +1,140 @@
-# Developer Guide
+# MeetingOrganizer - Developer Guide
+By: `AY1920S2-CS2113T-T12-1`
 
-## Design & Implementation
+## Table of Contents
+* [1. Setting up](#1-setting-up)
+* [2. Design](#2-design)
+	* [2.1. Architecture](#21-architecture)
+	* [2.2. UI component](#22-ui-component)
+	* [2.3. Module parsing logic component](#23-module-parsing-logic-component)
+	* [2.4. Member component](#24-member-component)
+	* [2.5. Meeting component](#25-model.meeting-component)
+	* [2.6. Exception classes](#26-exception-classes)
+	* [2.7. Common classes](#27-common-classes)
+* [3. Implementation](#3-implementation)
+	* [3.1. Detailed implementation of logic.modulelogic component](#31-detailed-implementation-of-logic.modulelogic-component)
+	* [3.2. Design Considerations](#32-design-considerations)
+* [4. Documentation](#4-documentation)
+* [5. Testing](#5-testing)
+* [6. Dev Ops](#6-dev-ops)
+* [Appendix A: Product Scope](#appendix-a-product-scope)
+	* [A.1. Target user profile](#a1-target-user-profile)
+	* [A.2. Value proposition](#a2-value-proposition)
+* [Appendix B: User Stories](#appendix-b-user-stories)
+* [Appendix C: Use Cases](#appendix-c-use-cases)
+* [Appendix D: Non-Functional Requirements](#appendix-d-non-functional-requirements)
+* [Appendix E: Glossary](#appendix-e-glossary)
+* [Appendix F: Product Survey](#appendix-f-product-survey)
+* [Appendix G: Instructions for Manual Testing](#appendix-g-instructions-for-manual-testing)
+	* [G.1. Launch and Shutdown](#g1-launch-and-shutdown)
+	* [G.2. Saving data](#g2-saving-data)
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+## 1. Setting up
 
+## 2. Design
 
-## Product Scope
-### Target user profile
+### 2.1. Architecture
+
+### 2.2. UI component
+
+### 2.3. Module parsing logic component
+The Module parsing logic component retrieves module information from NUSMODS API.
+
+**Structure of the module logic component:**
+<br>
+The module parsing logic of our application consists of 4 classes: ```TimetableParser ModuleApiParser ModuleHandler LessonsGenerator```.
+A detailed implementation would be explained in the implementation section.
+<br>
+ API: ```logic.modulelogic.LessonsGenerator```
+<br>
+1. ```LessonsGenerator``` uses the ```TimetableParser``` class to acquire the modules a user is taking, including the timeslots of those modules.
+2. ```LessonsGenerator``` also uses ```Modulehandler``` to retrieve a set of information related to a specific module.
+3. With both information, ```LessonsGenerator``` is able to dynamically generate the user's time-slots stored in ```ArrayList<String[]>``` via a series of Key-Value pair hashing.
+4. ```Arraylist<String[]> ``` contains the start/end time, days and weeks of all modules the user is taking.
```

Comment:
Consider giving the ArrayList<String[]> a name, or make it a separate class of its own, because referring to it as "ArrayList<String[]>" in the DG could be a bit confusing.